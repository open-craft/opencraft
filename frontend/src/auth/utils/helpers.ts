import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
