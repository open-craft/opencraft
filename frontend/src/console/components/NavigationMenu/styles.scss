.row.navigation-menu {
  align-items: center;
  text-align: center;
  border: 1px solid #c4c4c4;
  border-bottom: 1px solid #1a6ebb;
  padding-left: 10px;

  .col-md-1 {
    padding: 0;
  }

  .row.navigation-submenu {
    border: none;

    .col {
      display: inline-flex;
      justify-content: flex-end;
      align-items: center;

      .customizable-button {
        margin-left: 10px;
      }

      img {
        border: 1px solid #e7e7e7;
        margin-right: 10px;
        margin-left: 20px;
        border-radius: 5px;
        vertical-align: middle;
      }
    }
  }
}

.main-navigation-menu {
  margin-bottom: 30px;
  background: #ffffff;

  img {
    max-width: 100%;
  }
}
