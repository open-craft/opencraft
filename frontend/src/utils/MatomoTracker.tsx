import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
