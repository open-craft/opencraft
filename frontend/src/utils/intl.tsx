import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
