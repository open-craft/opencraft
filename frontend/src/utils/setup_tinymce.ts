import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
