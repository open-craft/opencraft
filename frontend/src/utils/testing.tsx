import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
