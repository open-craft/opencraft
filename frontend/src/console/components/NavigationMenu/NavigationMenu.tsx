import * as React from 'react';
// import messages from './displayMessages';
import './styles.scss';
import { Col, Row } from 'react-bootstrap';
import { ThemeSchema } from 'ocim-client';
import avatar from 'assets/avatar-default.png';
// import { CustomizableButton } from '../CustomizableButton';  # TODO: uncomment in BB-2219.
import { CustomizableLink } from '../CustomizableLink';
import { InstanceSettingsModel } from '../../models';

interface NavigationMenuProps {
  children?: React.ReactNode;
  instanceData: InstanceSettingsModel | null;
  themeData: ThemeSchema;
  loggedIn?: boolean;
}

export const NavigationMenu: React.FC<NavigationMenuProps> = (
  props: NavigationMenuProps
) => {
  const { themeData } = props;

  const userMenu = props.loggedIn ? (
    <Row className="navigation-menu navigation-submenu">
      <Col>
        <CustomizableLink linkColor={themeData.mainNavLinkColor} noHover>
          Help
        </CustomizableLink>
        <img src={avatar} alt="Avatar" />
        <CustomizableLink linkColor={themeData.mainNavLinkColor} noHover>
          JoeSoap
        </CustomizableLink>
        <CustomizableLink linkColor={themeData.userDropdownColor} noHover>
          <i className="fas fa-caret-down" />
        </CustomizableLink>
      </Col>
    </Row>
  ) : (
    <Row className="navigation-menu navigation-submenu">
      <Col>
        <CustomizableLink noHover>Register</CustomizableLink>
        <CustomizableLink noHover>Sign in</CustomizableLink>
        {/* TODO: Dummy buttons. While merging BB-2219 remove these from above and uncomment the ones below. */}
        {/* <CustomizableButton */}
        {/*  initialTextColor={themeData.btnRegisterColor} */}
        {/*  initialBackgroundColor={themeData.btnRegisterBg} */}
        {/*  initialBorderColor={themeData.btnRegisterBorderColor} */}
        {/*  initialHoverTextColor={themeData.btnRegisterHoverColor} */}
        {/*  initialHoverBackgroundColor={themeData.btnRegisterHoverBg} */}
        {/*  initialHoverBorderColor={themeData.btnRegisterHoverBorderColor} */}
        {/* > */}
        {/*  Register */}
        {/* </CustomizableButton> */}
        {/* <CustomizableButton */}
        {/*  initialTextColor={themeData.btnSignInColor} */}
        {/*  initialBackgroundColor={themeData.btnSignInBg} */}
        {/*  initialBorderColor={themeData.btnSignInBorderColor} */}
        {/*  initialHoverTextColor={themeData.btnSignInHoverColor} */}
        {/*  initialHoverBackgroundColor={themeData.btnSignInHoverBg} */}
        {/*  initialHoverBorderColor={themeData.btnSignInHoverBorderColor} */}
        {/* > */}
        {/*  Sign in */}
        {/* </CustomizableButton> */}
      </Col>
    </Row>
  );

  return (
    <Row
      className="main-navigation-menu navigation-menu"
      style={{ background: themeData.mainNavColor }}
    >
      <Col md={1}>
        {props.instanceData && props.instanceData.logo && (
          <img src={props.instanceData.logo} alt="Logo" />
        )}
      </Col>
      <Col md={1}>
        <CustomizableLink
          linkColor={themeData.mainNavLinkColor}
          borderBottomColor={themeData.mainNavItemBorderBottomColor}
          borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
          active={props.loggedIn}
        >
          Courses
        </CustomizableLink>
      </Col>
      <Col md={3}>
        <CustomizableLink
          linkColor={themeData.mainNavLinkColor}
          borderBottomColor={themeData.mainNavItemBorderBottomColor}
          borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
        >
          Discover New
        </CustomizableLink>
      </Col>
      <Col md={7}>{userMenu}</Col>
    </Row>
  );
};
