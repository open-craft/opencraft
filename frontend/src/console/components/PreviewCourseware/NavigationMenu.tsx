import * as React from 'react';
// import messages from './displayMessages';
import './styles.scss';
import { Col, Row } from 'react-bootstrap';
import { ThemeSchema } from 'ocim-client';
import { CustomizableLink } from '../CustomizableLink';
import { InstanceSettingsModel } from '../../models';

interface NavigationMenuProps {
  children?: React.ReactNode;
  instanceData: InstanceSettingsModel | null;
  themeData: ThemeSchema;
  loggedIn?: boolean;
}

export const NavigationMenu: React.FC<NavigationMenuProps> = (
  props: NavigationMenuProps
) => {
  const { themeData } = props;

  return (
      <div>

    <Row
    className="main-menu navigation-menu"
    style={{ background: themeData.headerBg }}
    >
        <Col md={1} className="logo">
        {props.instanceData && props.instanceData.logo && (
            <img src={props.instanceData.logo} alt="Logo" />
            )}
        </Col>
        <Col md={2.5} className="courseware-header">
          <Row className="org-name">
              edX: DemoX
          </Row>
          <Row>
              Demostration Course
          </Row>
        </Col>
        <Col md={2} className="courseware-header">
          <CustomizableLink
              linkColor={themeData.mainNavLinkColor}
              borderBottomColor={themeData.mainNavItemBorderBottomColor}
              borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
              >
              Discover New
          </CustomizableLink>
        </Col>
    </Row>
    <Row
      className="course-menu navigation-menu"
      style={{ background: themeData.headerBg }}
      >
      <Col md={1}>
        <CustomizableLink
          linkColor={themeData.mainNavLinkColor}
          borderBottomColor={themeData.mainNavItemBorderBottomColor}
          borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
          active={props.loggedIn}
          >
          Course
        </CustomizableLink>
      </Col>
      <Col md={1}>
        <CustomizableLink
          linkColor={themeData.mainNavLinkColor}
          borderBottomColor={themeData.mainNavItemBorderBottomColor}
          borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
          >
          Discussion
        </CustomizableLink>
      </Col>
      <Col md={1}>
        <CustomizableLink
          linkColor={themeData.mainNavLinkColor}
          borderBottomColor={themeData.mainNavItemBorderBottomColor}
          borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
          >
          Wiki
        </CustomizableLink>
      </Col>
      <Col md={1}>
        <CustomizableLink
          linkColor={themeData.mainNavLinkColor}
          borderBottomColor={themeData.mainNavItemBorderBottomColor}
          borderBottomHoverColor={themeData.mainNavItemHoverBorderBottomColor}
          >
          Progress
        </CustomizableLink>
      </Col>
    </Row>
            </div>
  );
};
