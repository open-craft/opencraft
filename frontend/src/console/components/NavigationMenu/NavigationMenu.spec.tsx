import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { NavigationMenu } from './NavigationMenu';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<NavigationMenu themeData={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
});
