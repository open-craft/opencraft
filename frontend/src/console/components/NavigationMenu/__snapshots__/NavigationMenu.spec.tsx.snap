// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="main-navigation-menu navigation-menu row"
  style={
    Object {
      "background": undefined,
    }
  }
>
  <div
    className="col-md-1"
  />
  <div
    className="col-md-1"
  >
    <button
      className="customizable-link"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderBottomColor": undefined,
          "color": undefined,
        }
      }
      type="button"
    >
      Courses
    </button>
  </div>
  <div
    className="col-md-3"
  >
    <button
      className="customizable-link"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderBottomColor": undefined,
          "color": undefined,
        }
      }
      type="button"
    >
      Discover New
    </button>
  </div>
  <div
    className="col-md-7"
  >
    <div
      className="navigation-menu navigation-submenu row"
    >
      <div
        className="col"
      >
        <button
          className="customizable-link noHover"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "borderBottomColor": undefined,
              "color": undefined,
            }
          }
          type="button"
        >
          Register
        </button>
        <button
          className="customizable-link noHover"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "borderBottomColor": undefined,
              "color": undefined,
            }
          }
          type="button"
        >
          Sign in
        </button>
      </div>
    </div>
  </div>
</div>
`;
